lib/pure/strutils.nim

```diff
@@ -1153,6 +1154,39 @@ func parseHexInt*(s: string): int {.rtl, extern: "nsuParseHexInt".} =
   if L != s.len or L == 0:
     raise newException(ValueError, "invalid hex integer: " & s)
 
+func parseBin*[T: SomeInteger](s: openArray[char], typ: typedesc[T]): T =
+  ## Parses a binary integer value contained in `s`.
+  ##
+  ## If `s` is not a valid binary integer, `ValueError` is raised. `s` can have
+  ## one of the following optional prefixes: `0b`, `0B`. Underscores within
+  ## `s` are ignored.
+  result = 0
+  let L = parseimpl.parseBinImpl(s, result, 0)
+  if L != s.len or L == 0:
+    raise newException(ValueError, "invalid binary integer: " & $s)
```

Comment:
use strbasics.add otherwise it shows as an array:
```
Error: unhandled exception: invalid binary integer: ['0', 'b', '1', '1', '1', '1', '1', '2'] [ValueError]
```

Reply:
Why not just keep the original form since we pass openArray[char]?